src/dmd/todt.d

```diff
@@ -553,7 +553,7 @@ extern (C++) void Expression_toDt(Expression e, ref DtBuilder dtb)
         if (auto sd = e.var.isSymbolDeclaration())
             if (sd.dsym)
             {
-                StructDeclaration_toDt(sd.dsym, dtb);
+                StructDeclaration_toDt(sd.dsym.isStructDeclaration(), dtb);
```

Comment:
This may call `StructDeclaration_toDt` with `null` if `sd.dsym` is a `ClassDeclaration`.
```suggestion
                if (auto struct_ = sd.dsym.isStructDeclaration())
                    StructDeclaration_toDt(struct_, dtb);
```

Reply:
That is not the case currently because it is always a `StructDeclaration`. 
Yes, DMD will need glue layer changes to support `__traits(initSymbol)` but that can wait until that is supported.

If you insist I can change this, but i think it is better to wait.

Reply:
Not strictly necessary, just a precaution against future changes because the type system is no longer enforces that `dsym` is  a `StructDeclaration`.